File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlCaseWhenClause.cs
namespace T1.SqlSharp.Expressions;

public class SqlWhenThenClause : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.WhenThen;
    public TextSpan Span { get; set; } = new();
    public required ISqlExpression When { get; set; }
    public required ISqlExpression Then { get; set; }

    public string ToSql()
    {
        var sql = "WHEN " + When.ToSql() + " THEN " + Then.ToSql();
        return sql;
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlCaseWhenClause.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlCaseWhenClause implements ISqlExpression {
    SqlType: SqlType = SqlType.WhenThen;
    Span: TextSpan = new TextSpan();
    When!: ISqlExpression;
    Then!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_WhenThen(this);
    }

    ToSql(): string {
        return `WHEN ${this.When.ToSql()} THEN ${this.Then.ToSql()}`;
    }
} 